### ISSUE TITLE
"Does not contain" and "Is not" filter also removes nulls

### ISSUE THREAD

**Issue by sbelak** [metabase/metabase#13332]
When applying a "does not contain" filter, it implicitly also filters away rows with null in that column. 

I think this is SQL behaviour, but not very nice UX.

**Comment by flamber** [comment]
Steps to reproduce:
1. Native question > Sample Dataset - and save as "Q1"
```
SELECT null AS "V"
UNION ALL
SELECT 'This has a value' AS "V"
```
2. Simple question > Saved Questions > Q1
![image](https://user-images.githubusercontent.com/1447303/94735735-7be63300-036b-11eb-9221-8ede09f3f7ec.png)
3. Filtered by "V" > "Is not" or "Does not contain" = `This has a value`
![image](https://user-images.githubusercontent.com/1447303/94735648-61ac5500-036b-11eb-860e-e2419027d5b9.png)

**Comment by nemanjaglumac** [comment]
Hi @sbelak (cc @robdaemon )
Sorry I have to reopen this issue, but there was [a repro](https://github.com/metabase/metabase/pull/13339) made for it, and it is still skipped. I pulled the latest changes, unskipped the test but it is still failing. :/

First of all, apologies if I could've done a better job of notifying the developer who was responsible for fixing this issue that there is a repro. We're all still getting used to this system.

So, please go to https://github.com/metabase/metabase/pull/13339, follow the steps to test this manually and tell me if it is failing for you as well. If it is, this issue was not resolved.

**Comment by nemanjaglumac** [comment]
As a general rule, when there is a repro for some issue, there is also a `.Reproduced` label assigned.
- You will always find the "history trace" with the mention, that links to that repro.
- I'm trying to write detailed description for every PR, and there is always a section called **"Now to test this manually"**

What I usually add in the PR description as well, but obviously forgot in this one is a line similar to:
> When the issue is fixed, unskip the test and merge it together with the issue.

Please let me know if (and how) I can make it easier for you guys, until we all get used to this. 🙏

**Comment by sbelak** [comment]
The repro actually finds a different bug -- the FE gives you "no results" if the dataset returned is `[[null]]`.

**Comment by flamber** [comment]
That FE handling sounds vaguely similar to #5419.
I guess the takeaway here is to remove this line and unskip the repro, then both issue and test are working, and can be closed:
https://github.com/metabase/metabase/pull/13339/files#diff-63235caab0be15e7fb2ea15b1c36af3264eef31e8fb7cc6caaa7a822b91a242cR264

**Comment by sbelak** [comment]
I've fixed the Cypress in #13570 by selecting an additional dummy col.

**Comment by sbelak** [comment]
Fixed via #13570
